src/abb_system_position_only.cpp

```diff
@@ -13,242 +13,200 @@
 // limitations under the License.
 
 #include "ros2_control_abb_driver/abb_system_position_only.hpp"
-
-#include <chrono>
-#include <cmath>
-#include <limits>
-#include <memory>
-#include <thread>
-#include <vector>
-
-#include "abb_libegm/egm_controller_interface.h"
-#include "hardware_interface/types/hardware_interface_type_values.hpp"
-#include "rclcpp/rclcpp.hpp"
+#include "ros2_control_abb_driver/utilities.hpp"
 
 using namespace std::chrono_literals;
 
 namespace ros2_control_abb_driver
 {
 static constexpr size_t NUM_CONNECTION_TRIES = 100;
-static constexpr auto DEG_TO_RAD = 180 / M_PI;
+static const rclcpp::Logger LOGGER = rclcpp::get_logger("ABBSystemPositionOnlyHardware");
 
 CallbackReturn ABBSystemPositionOnlyHardware::on_init(const hardware_interface::HardwareInfo & info)
 {
-  RCLCPP_INFO(rclcpp::get_logger("ABBSystemPositionOnlyHardware"), "on_init()");
+  RCLCPP_INFO(LOGGER, "on_init()");
 
   if (hardware_interface::SystemInterface::on_init(info) != CallbackReturn::SUCCESS) {
     return CallbackReturn::ERROR;
   }
 
-  hw_states_.resize(info_.joints.size(), std::numeric_limits<double>::quiet_NaN());
-  hw_commands_.resize(info_.joints.size(), std::numeric_limits<double>::quiet_NaN());
-  robotstudio_port_ = stod(info_.hardware_parameters["robotstudio_port"]);
+  // TODO(seng):
+  // Rename robotstudio_port to egm_port
+  // Create new hardware parameter for:
+  //  - rws_port (8000)
```

Comment:
`tcp://:80` is used by the RWS server.

Your driver is the client.

Which port you use locally does not matter to the server.

Reply:
I agree with your statements, but I'd like to choose a default local port number and I'm wondering what local port would be simplest to use. 

My question isn't really a big deal, as ideally a user will change the IP and port to match their setup, but it would be nice to have sane default values. I can just leave it the local port as 80, unless you have any objections. This would work for people that configure their virtual controller whitelist for their local machine, but may cause issue as it's a reserved port on Linux.

Reply:
I'm not sure I understand what you're writing.

The IRC5 controller is the server. The RWS HTTP server running on the IRC uses `tcp://:80` to accept requests from clients.

Your driver is one of those clients.

The client will use a random, local, TCP port, assigned by the OS, to connect to the server.

There should be no need to "choose a default local port number", your OS will choose one for you.

Reply:
I wasn't able to get the whitelisting to work for some reason, so I've been using port-forwarding as a workaround. I understand your point that port-forwarding shouldn't be the default - I've changed the code such that `rws_ip` is specified from a launch parameter, as it should never be localhost if whitelisting is working correctly.

Appreciate your help!